System error: PhpParser\NodeAbstract::getComments()

Hello, I just got my first issue with Rector 👍 

PHP 8.4.15
Rector version: 2.2.14

Report:
```
[ERROR] Could not process
         "/Users/gilles/PhpstormProjects/automate-php/src/Controller/RevueController.php" file, due to:
         "System error: "PhpParser\NodeAbstract::getComments(): Return value must be of type array, true returned"

         Stack trace:
         #0 /Users/gilles/.composer/vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeAbstract.php(119):
         PhpParser\NodeAbstract->getComments()
         #1 /Users/gilles/.composer/vendor/rector/rector/src/BetterPhpDocParser/PhpDocInfo/PhpDocInfoFactory.php(77):
         PhpParser\NodeAbstract->getDocComment()
         #2 /Users/gilles/.composer/vendor/rector/rector/rules/Php80/Rector/Class_/AnnotationToAttributeRector.php(152):
         Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory->createFromNode(Object(PhpParser\Node\Expr\ArrowFunctio
         n))
         #3 /Users/gilles/.composer/vendor/rector/rector/src/Rector/AbstractRector.php(115):
         Rector\Php80\Rector\Class_\AnnotationToAttributeRector->refactor(Object(PhpParser\Node\Expr\ArrowFunction))
         #4
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(91)
         : Rector\Rector\AbstractRector->enterNode(Object(PhpParser\Node\Expr\ArrowFunction))
         #5
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187
         ): Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Arg))
         #6
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78)
         : Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray(Array)
         #7
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(114
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Expr\FuncCal
         l))
         #8
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(114
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Expr\Assign)
         )
         #9
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Express
         ion))
         #10
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78)
         : Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray(Array)
         #11
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Foreach
         _))
         #12
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78)
         : Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray(Array)
         #13
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Foreach
         _))
         #14
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78)
         : Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray(Array)
         #15
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\If_))
         #16
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78)
         : Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray(Array)
         #17
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\ClassMe
         thod))
         #18
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78)
         : Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray(Array)
         #19
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Class_)
         )
         #20
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78)
         : Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray(Array)
         #21
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187
         ):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode(Object(PhpParser\Node\Stmt\Namespa
         ce_))
         #22
         /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(60)
         : Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray(Array)
         #23 /Users/gilles/.composer/vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(51):
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverse(Array)
         #24 /Users/gilles/.composer/vendor/rector/rector/src/Application/FileProcessor.php(95):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverse(Array)
         #25 /Users/gilles/.composer/vendor/rector/rector/src/Application/ApplicationFileProcessor.php(178):
         Rector\Application\FileProcessor->processFile(Object(Rector\ValueObject\Application\File),
         Object(Rector\ValueObject\Configuration))
         #26 /Users/gilles/.composer/vendor/rector/rector/src/Application/ApplicationFileProcessor.php(152):
         Rector\Application\ApplicationFileProcessor->processFile(Object(Rector\ValueObject\Application\File),
         Object(Rector\ValueObject\Configuration))
         #27 /Users/gilles/.composer/vendor/rector/rector/src/Application/ApplicationFileProcessor.php(128):
         Rector\Application\ApplicationFileProcessor->processFiles(Array, Object(Rector\ValueObject\Configuration),
         Object(Closure), Object(Closure))
         #28 /Users/gilles/.composer/vendor/rector/rector/src/Console/Command/ProcessCommand.php(182):
         Rector\Application\ApplicationFileProcessor->run(Object(Rector\ValueObject\Configuration),
         Object(RectorPrefix202512\Symfony\Component\Console\Input\ArgvInput))
         #29 /Users/gilles/.composer/vendor/rector/rector/vendor/symfony/console/Command/Command.php(289):
         Rector\Console\Command\ProcessCommand->execute(Object(RectorPrefix202512\Symfony\Component\Console\Input\ArgvI
         nput), Object(RectorPrefix202512\Symfony\Component\Console\Output\ConsoleOutput))
         #30 /Users/gilles/.composer/vendor/rector/rector/vendor/symfony/console/Application.php(899):
         RectorPrefix202512\Symfony\Component\Console\Command\Command->run(Object(RectorPrefix202512\Symfony\Component\
         Console\Input\ArgvInput), Object(RectorPrefix202512\Symfony\Component\Console\Output\ConsoleOutput))
         #31 /Users/gilles/.composer/vendor/rector/rector/vendor/symfony/console/Application.php(279):
         RectorPrefix202512\Symfony\Component\Console\Application->doRunCommand(Object(Rector\Console\Command\ProcessCo
         mmand), Object(RectorPrefix202512\Symfony\Component\Console\Input\ArgvInput),
         Object(RectorPrefix202512\Symfony\Component\Console\Output\ConsoleOutput))
         #32 /Users/gilles/.composer/vendor/rector/rector/src/Console/ConsoleApplication.php(60):
         RectorPrefix202512\Symfony\Component\Console\Application->doRun(Object(RectorPrefix202512\Symfony\Component\Co
         nsole\Input\ArgvInput), Object(RectorPrefix202512\Symfony\Component\Console\Output\ConsoleOutput))
         #33 /Users/gilles/.composer/vendor/rector/rector/vendor/symfony/console/Application.php(162):
         Rector\Console\ConsoleApplication->doRun(Object(RectorPrefix202512\Symfony\Component\Console\Input\ArgvInput),
         Object(RectorPrefix202512\Symfony\Component\Console\Output\ConsoleOutput))
         #34 /Users/gilles/.composer/vendor/rector/rector/bin/rector.php(130):
         RectorPrefix202512\Symfony\Component\Console\Application->run()
         #35 /Users/gilles/.composer/vendor/rector/rector/bin/rector(5): require_once('/Users/gilles/....')
         #36 /Users/gilles/.composer/vendor/bin/rector(119): include('/Users/gilles/....')
         #37 {main}". On line: 110
```

--- a/rules/Php74/Rector/Closure/ClosureToArrowFunctionRector.php
+++ b/rules/Php74/Rector/Closure/ClosureToArrowFunctionRector.php
@@ -89,7 +89,7 @@ CODE_SAMPLE
         $comments = $node->stmts[0]->getAttribute(AttributeKey::COMMENTS) ?? [];
         if ($comments !== []) {
             $this->mirrorComments($arrowFunction->expr, $node->stmts[0]);
-            $arrowFunction->setAttribute(AttributeKey::COMMENTS, true);
+            $arrowFunction->setAttribute(AttributeKey::COMMENTS, []);
         }
 
         return $arrowFunction;

--- a/rules/Php74/Rector/Closure/ClosureToArrowFunctionRector.php
+++ b/rules/Php74/Rector/Closure/ClosureToArrowFunctionRector.php
@@ -89,7 +89,7 @@ CODE_SAMPLE
         $comments = $node->stmts[0]->getAttribute(AttributeKey::COMMENTS) ?? [];
         if ($comments !== []) {
             $this->mirrorComments($arrowFunction->expr, $node->stmts[0]);
-            $arrowFunction->setAttribute(AttributeKey::COMMENTS, []);
+            $arrowFunction->setAttribute(AttributeKey::COMMENTS, $node->stmts[0]->getComments());
         }
 
         return $arrowFunction;

